e2e-polybft/cardanofw/test-apex-bridge.go

```diff
@@ -4,265 +4,70 @@ import (
 	"context"
 	"fmt"
 	"os"
-	"path/filepath"
 	"testing"
-	"time"
 
-	"github.com/0xPolygon/polygon-edge/consensus/polybft/wallet"
-	"github.com/0xPolygon/polygon-edge/helper/common"
-	"github.com/0xPolygon/polygon-edge/txrelayer"
-	"github.com/0xPolygon/polygon-edge/types"
-	cardanowallet "github.com/Ethernal-Tech/cardano-infrastructure/wallet"
 	"github.com/stretchr/testify/require"
-	"github.com/umbracle/ethgo"
 )
 
 const FundTokenAmount = uint64(100_000_000_000)
 
-func RunCardanoCluster(
-	t *testing.T,
-	ctx context.Context,
-	config *RunCardanoClusterConfig,
-) (*TestCardanoCluster, error) {
-	t.Helper()
-
-	networkMagic := GetNetworkMagic(config.NetworkType)
-	networkName := GetNetworkName(config.NetworkType)
-	logsDir := fmt.Sprintf("%s/%d", getCardanoBaseLogsDir(t, networkName), config.ID)
-
-	if err := common.CreateDirSafe(logsDir, 0750); err != nil {
-		return nil, err
-	}
-
-	cluster, err := NewCardanoTestCluster(t,
-		WithID(config.ID+1),
-		WithNodesCount(config.NodesCount),
-		WithStartTimeDelay(time.Second*5),
-		WithPort(5100+config.ID*100),
-		WithOgmiosPort(1337+config.ID),
-		WithLogsDir(logsDir),
-		WithNetworkMagic(networkMagic),
-		WithNetworkType(config.NetworkType),
-		WithConfigGenesisDir(networkName),
-		WithInitialFunds(config.InitialFundsKeys, config.InitialFundsAmount),
-	)
-	if err != nil {
-		return nil, err
-	}
-
-	fmt.Printf("Waiting for sockets to be ready\n")
-
-	if err := cluster.WaitForReady(time.Minute * 2); err != nil {
-		return nil, err
-	}
-
-	if err := cluster.StartOgmios(t, config.ID); err != nil {
-		return nil, err
-	}
-
-	if err := cluster.WaitForBlockWithState(10, time.Second*120); err != nil {
-		return nil, err
-	}
-
-	fmt.Printf("Cluster %d is ready\n", config.ID)
-
-	return cluster, nil
-}
-
 func SetupAndRunApexBridge(
 	t *testing.T,
 	ctx context.Context,
-	dataDir string,
-	apexSystem *ApexSystem,
-) *TestCardanoBridge {
+	opts ...ApexSystemOptions,
+) *ApexSystem {
 	t.Helper()
 
-	const (
-		bladeEpochSize = 5
-		numOfRetries   = 90
-		waitTime       = time.Second * 2
-	)
+	bridgeDataDir := "../../e2e-bridge-data-tmp-" + t.Name() // path.Join(path.Dir(primeCluster.Config.TmpDir), "bridge")
 
-	cleanupDataDir := func() {
-		os.RemoveAll(dataDir)
-	}
+	os.RemoveAll(bridgeDataDir)
 
-	cleanupDataDir()
+	apexSystem := NewApexSystem(bridgeDataDir, opts...)
 
-	cb := NewTestCardanoBridge(dataDir, apexSystem.Config)
+	fmt.Printf("Starting chains...\n")
 
-	require.NoError(t, cb.CreateCardanoWallets())
+	apexSystem.StartChains(t)
 
-	fmt.Printf("Wallets and addresses created\n")
+	fmt.Printf("Chains have been started. Starting bridge chain...\n")
 
-	cb.StartValidators(t, bladeEpochSize)
+	apexSystem.StartBridgeChain(t)
 
-	fmt.Printf("Validators started\n")
+	fmt.Printf("Bridge chain has been started\n")
```

Comment:
Since apexSystem.StartBridgeChain(t) waits for validators to be ready, this printf should be fmt.Printf("Bridge chain has been started. Validators are ready.\n")